app/controllers/accounts_controller.rb

```diff
@@ -1,5 +1,8 @@
 class AccountsController < ApplicationController
   def show
-    @account = current_user.account
+    @account = params[:id] ? Account.find(params[:id]) : current_user.account
+    authorize @account, :show?
+    session[:account_id] = @account.id
```

Comment:
I think you can use https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html to make the access to the current account and current user's account more accessible. e.g:

```
Current.current_user_account
Current.current_account
```